File: src/app/data/flight.ts
import { Airline} from "./airline";

export interface Flight {
  flightId:string;
  departureAirport:string;
  arrivalAirport:string;
  departureDate:Date;
  arrivalDate:Date;
  departureTime:string;
  arrivalTime:string;
  airline:Airline;
}

export type FlightCreateInput = Omit<Flight,'flightId' |'airline' | 'departureAirport' | 'arrivalAirport' | 'departureDate'
 | 'arrivalDate' | 'departureTime' | 'arrivalTime'>&{
  airlineId:string;
}

File: src/main/java/fr/joellehuyen/reflights/models/Flight.java
package fr.joellehuyen.reflights.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name="flight")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Flight {
    @Id
    private String id;
    private String departureAirport;
    private String arrivalAirport;
    private LocalDate departureDate;
    private LocalDate arrivalDate;
    private LocalTime departureTime;
    private LocalTime arrivalTime;

    @ManyToOne
    @JoinColumn(name="airline_id", nullable=false)
    private Airline airline;
}
